frontend/store/collaterals.ts

```diff
@@ -69,6 +80,62 @@ export const actions = {
             commit('updateCollateral', updated);
         }
     },
+    async fetchCollateralStatuses({ commit, dispatch, rootGetters }: ActionContext<State, State>) {
+        const network = rootGetters['network/getMakerNetwork'];
+        if (!network) {
+            return;
+        }
+        for (const collateral of Object.values(COLLATERALS)) {
+            const tokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateral.symbol).catch(() => {
+                return undefined;
+            });
```

Comment:
Why do you need to implement collateral fetching 2 times? Can't you simply call `fetchCollateralStatus` here in the loop?